src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -298,38 +306,79 @@ public void schedulePeriodicFleetStatusDataUpdate(boolean isDuringConnectionResu
     @SuppressWarnings("PMD.UnusedFormalParameter")
     private void handleServiceStateChange(GreengrassService greengrassService, State oldState,
                                           State newState) {
-        // Not reporting status of other components in errored component message because a deployment could
-        // be in-progress and other component's fleet config arn may not have updated
         if (newState.equals(State.ERRORED)) {
-            Set<GreengrassService> erroredComponentSet = new HashSet<>();
-            erroredComponentSet.add(greengrassService);
-
             // Evaluate overall status based on current state of all components
             Instant now = Instant.now();
             AtomicReference<OverallStatus> overAllStatus = new AtomicReference<>(OverallStatus.HEALTHY);
             this.kernel.orderedDependencies().forEach(service -> {
                 serviceFssTracksMap.put(service, now);
                 overAllStatus.set(getOverallStatusBasedOnServiceState(overAllStatus.get(), service));
             });
-
-            uploadFleetStatusServiceData(erroredComponentSet, overAllStatus.get(), null,
+            // Not reporting status of other components in errored component message because a deployment could
+            // be in-progress and other component's fleet config arn may not have updated
+            Set<GreengrassService> erroredComponent = new HashSet<>();
+            erroredComponent.add(greengrassService);
+            if (isDeploymentInProgress.get()) {
+                uploadFleetStatusServiceData(erroredComponent, overAllStatus.get(), null,
```

Comment:
since you are not using erroredComponentSet anywhere else in this method, can you create an inline set in the call to `uploadFleetStatusServiceData`?

like 
```
uploadFleetStatusServiceData(Collections.singleton(greengrassService)), ...
```

Reply:
Updated